tasks/tomcat_wars.yml

```diff
@@ -0,0 +1,29 @@
+---
+- name: "Ensure webapp dir exists (url)"
+  shell: "mkdir -p {{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps"
+  when:
+    - "tomcat_wars.url is defined"
+  with_items: "{{ tomcat_wars.url }}"
+- name: "loop over tomcat_wars"
+  get_url:
+    dest: "{{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps/"
+    url: "{{ item }}"
+    validate_certs: false
+  when:
+    - "tomcat_wars.url is defined"
+  with_items: "{{ tomcat_wars.url }}"
+- name: "Ensure webapp dir exists (s3)"
+  shell: "mkdir -p {{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps"
+  when:
+    - "tomcat_wars.s3 is defined"
+  with_items: "{{ tomcat_wars.s3 }}"
+- name: "loop over tomcat_wars in s3"
+  aws_s3:
```

Comment:
Great module! I think it's probably easier to refer to the s3 object in variables, something like this:
```
tomcat_layout:
  - name: appone
    wars:
      - url: https://tomcat.apache.org/tomcat-6.0-doc/appdev/sample/sample.war
      - s3:
        - bucket: mybucket
          object: /my/desired/key.txt
```

(I'm not sure about the suggested layout, seems a bit complex...) I'm open to suggestions.

Reply:
Thats the exact layout I built for this. :)